src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,931 @@
+      random = new Random(seed);
+      addUsers(userNames);
+    }
+
+    private String getNextOncokbEndpoint() {
+      return sampleOncokbEndpoints[Math.abs(
+          random.nextInt() % sampleOncokbEndpoints.length
+        )];
+    }
+
+    /**
+     * Get value from files JSON object based on the JSON path created by the
+     * {@code keys}. If an object is missing in the JSON path then it's created.
+     * @param keys The keys in JSON path
+     * @return The resulting {@code KeyValuePair}
+     */
+    private KeyValuePair<String, JsonObject> fetchJsonObjectFromFilesObject(
+      String... keys
+    ) {
+      JsonObject obj = files;
+      for (int i = 0; i < keys.length - 1; i++) {
+        String key = keys[i];
+        if (!obj.has(key)) {
+          obj.add(key, new JsonObject());
+        }
+        obj = (JsonObject) obj.get(key);
+      }
+      String lastKey = keys[keys.length - 1];
+      return new KeyValuePair<>(lastKey, obj);
+    }
+
+    /**
+     * Get value from files JSON object based on the JSON path created by the
+     * {@code filePath}/[{@code index}]/{@code keys}.
+     * If an object is missing in the JSON path then it's created.
+     * @param filePath First key in the files object.
+     * @param index Array index of the array the {@code filePath} parameter points to.
+     * @param keys The remaining JSON path keys.
+     * @return The resulting {@code KeyValuePair}
+     */
+    private KeyValuePair<String, JsonObject> fetchJsonObjectFromFilesObject(
+      String filePath,
+      int index,
+      String... keys
+    ) {
+      JsonObject obj = files;
+      if (!obj.has(filePath)) {
+        obj.add(filePath, new JsonArray());
+      }
+      JsonArray arr = (JsonArray) obj.get(filePath);
+      while (index >= arr.size()) {
+        arr.add(new JsonObject());
+      }
+      obj = (JsonObject) arr.get(index);
+
+      for (int i = 0; i < keys.length - 1; i++) {
+        String key = keys[i];
+        if (!obj.has(key)) {
+          obj.add(key, new JsonObject());
+        }
+        obj = (JsonObject) obj.get(key);
+      }
+      String lastKey = keys[keys.length - 1];
+      if (!obj.has(lastKey)) {
+        obj.addProperty(lastKey, 0);
+      }
+      return new KeyValuePair<>(lastKey, obj);
+    }
+
+    /**
+     * Adds the passed value to the value found in the files JSON object based on the JSON path created by the
+     * keys. If an object is missing in the JSON path then it's created.
+     * @param value Adds the specified value at the given JSON path.
+     * @param keys The keys in JSON path.
+     */
+    private void safeAddNestedValueInFilesObject(int value, String... keys) {
+      KeyValuePair<String, JsonObject> pair = fetchJsonObjectFromFilesObject(
+        keys
+      );
+      String lastKey = pair.key;
+      JsonObject obj = pair.value;
+      if (!obj.has(lastKey)) {
+        obj.addProperty(lastKey, 0);
+      }
```

Comment:
Why isn't this part of `fetchJsonObjectFromFilesObject`? The other `fetchJsonObjectFromFilesObject` method does have it.